### ISSUE TITLE
disproportionate image (loosing apsect ratio)

### ISSUE THREAD

**Issue by angelotumedei** [www-smithchain-com/Apex-Camera-Extension-ACE#6]
If you shoot with the mobile phone vertically, the image is enlarged and loses aspect ratio, if instead the screen is horizontally it does not happen (in fact the camera from the PC shoots well). How to fix it?
Thank you, Angelo

**Comment by angelotumedei** [comment]
I also did the test on your demo site and the problem is the same: you lose aspect ratio if you hold the phone vertically.
Thank you again, Angelo

**Comment by angelotumedei** [comment]
hello, is there any news?
Thank you, Angelo

**Comment by angelotumedei** [comment]
For now I solved in this way:

Modified the CamIntegration.js file:
Line 47:
 //var div_rgn = $('<div style="text-align:center;" class="image-capture-head"> <video id="video"  style="width:'+p_cam_width+';height:'+p_cam_height+';" autoplay></video> <canvas id="canvas" width="640" height="480" style="display:none;"></canvas></div> <div style="text-align:center;"><button type="button" id="save-image-capture" class="t-Button t-Button--primary">'+p_btn_lable+'</button><button type="button" id="save_back" class="t-Button t-Button--primary">'+p_btn_switch+'</button></div> ');

 var div_rgn = $('<div style="text-align:center;" class="image-capture-head"> <video id="video"  style="width:'+p_cam_width+';height:'+p_cam_height+';" autoplay></video> <canvas id="canvas" width="720" height="1280" style="display:none;"></canvas></div> <div style="text-align:center;"><button type="button" id="save-image-capture" class="t-Button t-Button--primary">'+p_btn_lable+'</button><button type="button" id="save_back" class="t-Button t-Button--primary">'+p_btn_switch+'</button></div> ');

line 170:
//                                        context.drawImage(video, 0, 0, 640, 480);
                                        context.drawImage(video, 0, 0, 720, 1280);

basically I changed the fixed resolutions and I inverted them.

Now if you take a photo with the mobile phone held vertically, the image is good and this is the normal behavior of users.

- I did well?
- are there any other methods?
- Were you able to figure out how to make the rear camera the default one?

Hi, Angelo

**Comment by mcuambe** [comment]
Hi everyone,
I tried to use this Plugin but when try to take picture it raises an error below

Error: Application 16164 Dialog page 14 cannot be rendered successfully.  Ensure the page template in use on page 14 is of template type "Dialog page", with appropriate JavaScript dialog initialization, dialog closure and dialog cancel code defined.

P.s I'm using a dilog modal page

**Comment by vikasapexian** [comment]
Ok .
Let me check.
